src/Siganberg.SerilogElasticSearch/Utilities/EnrichHelper.cs

```diff
@@ -18,37 +16,42 @@ internal static void AddDiagnosticContext(RequestLoggingOptions options, IConfig
             async void OptionsEnrichDiagnosticContext(IDiagnosticContext diagnosticContext, HttpContext httpContext)
             {
                 if (httpContext?.Request == null) return;
-                try
-                {
-                    diagnosticContext.Set("RequestBody", await ReadRequestBody(httpContext.Request));
-                    diagnosticContext.Set("Path", httpContext.Request.Path);
-                    diagnosticContext.Set("QueryString", httpContext.Request.QueryString);
-                    var includeRequestHeaders = config["Serilog:RequestLoggingOptions:IncludeRequestHeaders"];
-                    if (string.IsNullOrWhiteSpace(includeRequestHeaders) || includeRequestHeaders.ToLower() == "true") diagnosticContext.Set("RequestHeaders", FormatHeader(httpContext.Request.Headers, config));
-                } 
-                catch (Exception e)
-                {
-                    var factory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
```

Comment:
If you had an empty HTTP Request Body, an InvalidOperationException would be thrown from the `ReadRequestBody` method. THEN, inside this exception handler, an ObjectDisposedException would be thrown on this line. So I decided to handle any potential exceptions inside the `ReadRequestBody` method since that's where the brittleness resides. 

If an exception does occur attempting to read the body, this is the log message that will be output. It's not exact, but it's better than nothing.

```JSON
{"time" : "5/31/2022 11:09:21 PM", "level" : "ERROR", "message" : "The connection was aborted by the application." , "exception" : "Microsoft.AspNetCore.Connections.ConnectionAbortedException: The connection was aborted by the application.\r\n ---> System.InvalidOperationException: Reading is already in progress.\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1ContentLengthMessageBody.VerifyIsNotReading()\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1ContentLengthMessageBody.TryReadInternal(ReadResult& readResult)\r\n   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1MessageBody.OnConsumeAsync()\r\n   --- End of inner exception stack trace ---", "callsite" : "Microsoft.AspNetCore.Server.Kestrel"}
```